### ISSUE TITLE
1.3.0.6 and Alt + Tab switching from SuperPutty - only double one works

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jimradford/superputty#216]
```
This is not duplicate of #215 bug!

What steps will reproduce the problem?
1. To run SuperPutty and to run one session for example
2. Now if i switch Alt + Tab i see again SuperPutty but without focus in 
current session. I should press again Alt + Tab for switching to other Windows 
application.

What is the expected output? What do you see instead?

First time pressing of Alt + Tab should switch to other Windows application. 
But should be done double Alt + Tab pressing for it

What version of the product are you using? On what operating system?

OS is Windows 7 (may be affected other OSes)

```

Original issue reported on code.google.com by `perlo...@gmail.com` on 27 Aug 2012 at 3:47

**Comment by biggianteye** [comment]
This still exists in 1.4.0.6. I am using 64-bit Windows 8.1.

**Comment by kitsirota** [comment]
+1 This is a problem in Windows 10 Pro

I've also tried all of the focus activator variants in the options menu with no luck.  

alt-tab to switch to superputty
alt-tab-tab to switch from superputty to the previous tab

This is pretty annoying, any help would be greatly appreciated.

Thanks!

**Comment by shallot** [comment]
I have also been reproducing the same issue on Windows 8.1, usually in combination with Firefox.

The upgrade to Windows 10 changed the behavior somewhat - it's now apparent that SuperPutty is doing something strange: when you press Alt+Tab within SuperPutty, you see in the popup window that the SuperPutty window is still the one selected, and you have to press Tab another time to get it to move to the next window. But with any other Windows program, this happens by default, you press Alt+Tab only once and it instantly moves to the next window.

**Comment by jimradford** [comment]
@shallot I believe what you describe is due to SuperPuTTY is one running application, and putty inside is another, So the first ALT+TAB switches from the internal process (putty for example) and puts the focus on SuperPutty, the Second TAB changes focus to the next application running, so ALT+TAB+TAB gets you out of SuperPuTTY and Alt+Tab gets you back. of course 1  alt+tab should bring back SuperPuTTY and the last active window should be in focus.

That all being said, I am not sure how/if I can change that behavior since its Windows itself seeing the two applications running even though only one shows. I'll dig into this a little more and see if there is a simple or a hard solution.

**Comment by MichaelThessel** [comment]
I wanted to like it but this is a deal breaker. I assume there has been no progress on this?

**Comment by miltosc** [comment]
Same problem here. Windows 8 Pro x64. It is really annoying! It needs 2 times to pres alt + tab to switch windows.

**Comment by boblarouche** [comment]
Same here, running Windows 10 x64. I like the software, but this is just annoying... please update the alt + tab tab issue. Many thanks to you.

**Comment by roltd** [comment]
**boblarouche**
I have the same issue in the win10pro 64bit.

**Comment by jzakarias** [comment]
Same issue here on Win 10 Pro using v1.4.0.8

**Comment by Racer84** [comment]
Here is simple autohotkey script that helps to live with the issue:

`
LAltTab()
{
    Send {Alt down}{tab 2}{Alt up}
}

SetKeyDelay, 80
#IfWinActive ahk_class PuTTY                                                                                        
LAlt & Tab::LAltTab()
#IfWinActive
`

**Comment by vai221094** [comment]
Could you

**Comment by vai221094** [comment]
Could you please help with the steps on how to use the above script you mentioned, please ?

**Comment by KeithTurkowski** [comment]
Nice, script, but you should leave off {Alt up} Like this:

`LAltTab()
{
Send {Alt down}{tab 2}
}

SetKeyDelay, 80
#IfWinActive ahk_class PuTTY
LAlt & Tab::LAltTab()
#IfWinActive`

You can use this script by installing autohotkey and then pasting the code into a text file with the *.ahk extension.

**Comment by Chealer** [comment]
Why is this ticket closed? I found it only because of #743 .

**Comment by sgatade** [comment]
Switche to MobaXTerm...

Feeling normal again!

Regards,

SG

**Comment by larsen0815** [comment]
The autohotkey scripts works, but normally the next Alt+Tab would return to the previous program, which it does not with the script.
- SuperPuTTY is active
- Press Alt+Tab -> Active window now e.g. is Chrome
- Pressing Alt+Tab again...

Expected:
- SuperPuTTY is the first program to switch to

Actual:
- SuperPuTTY is the last program in line. Would need Shift to go reverse

**Comment by Chealer** [comment]
I am afraid MobaXTerm is a proprietary product :-/

As this ticket has been closed for way too long, I filed #809 to track the same issue.

**Comment by clarkwang** [comment]
@Chealer 

Mobaxterm's [license](https://mobaxterm.mobatek.net/license.html) allows use in commercial environment:

> Individual end-user is allowed to download (only from MobaXterm website: https://mobaxterm.mobatek.net) and to use MobaXterm Home Edition **in a commercial or company environment**. However, **software installation must be performed by the end-user himself**: the user who uses MobaXterm Home Edition inside a company must be the same person who downloaded the software and installed it. It is therefore not allowed to redistribute or deploy MobaXterm Home Edition inside a company. It is also **not allowed for multiple users to use a single shared installation** of MobaXterm Home Edition in a company, whether at the same time or not. These usages are covered by MobaXterm Professional Edition.

For me the only issue is the Home Edition only supports 12 sessions.

**Comment by fitzg2** [comment]
Still present 9 years later. Windows 10

**Comment by jimradford** [comment]
@fitzg2 I don't know that this can be fixed without getting rid of putty and using some other ssh library internally. Putty and SuperPuTTY are different processes.
